hotel-bookings-api/server.js

```diff
@@ -15,6 +16,66 @@ app.get("/", function (request, response) {
 
 // TODO add your routes and helper functions here
 
-const listener = app.listen(process.env.PORT, function () {
+// all the bookings
+app.get("/bookings", function (request, response) {
+  response.send(bookings);
+});
+
+// creating new booking
+app.post("/bookings", function (request, response) {
+  // reject booking if ant object field is empty - simple validation
+  
+   const booking = request.body;
+  if (
+    (!booking.name || !booking.title || !booking.
+    firstName ||
+      !booking.surname ||
+      !booking.email ||
+      !booking.roomId ||
+      !booking.checkInDate ||
+      !booking.checkOutDate)
+  ) {
+    return response.status(400).send("invalid request..missing data");
+  }
+
+  let max = bookings.reduce((a, b) => {
+    // max items in our array
+    return { id: Math.max(a.id, b.id) };
+  });
+ 
+  booking.id = max.id + 1; // id will be server generater
+  bookings.push(booking); //pushing our new booking into bookings
+  response.json({
+    data: booking,
+    message: "booking created",
+  });
+});
+
+// getting booking by ID
+app.get("/bookings/:id", function (request, response) {
+  const bookingId = bookings.find((booking) => booking.id == request.params.id);
+
+  if (!bookingId) {
+    response.status(404).send("booking not found");
+  } else {
+    response.json(bookingId);
+  }
+});
+
+// delete
+
+app.delete("/bookings/:id", function (request, response) {
```

Comment:
The code you've written is excellent! You've demonstrated a good command and understanding of Restful API creation with Node. As a suggestion, and you will see this in most professional (enterprise-level) codebases, is that devs usually like to separate their logic in different folders/directories/packages. So it's easier for the eye, and better to maintain in the long run. 

So what you can do here is, take all the functions that you define to perform an action (delete, get, post), and by functions I refer to these:
```
function (request, response) {
 const index = bookings.findIndex(
    (booking) => booking.id == request.params.id
  );
  if (index == -1) {
    response.status(404).send("Booking not found");
  } else {
    bookings.splice(index, 1);
    response.send("Booking deleted");
  }
}

```

And put them in a different directory called **controllers** and import them in the `server.js` file. This way, you have a cleaner looking codebase. 

You can read more on that [here](https://dev.to/ericchapman/nodejs-express-part-5-routes-and-controllers-55d3)